### ISSUE TITLE
[BUG] Properties and methods are being duplicate through inheritance

### ISSUE THREAD

**Issue by armoucar** [compodoc/compodoc#1034]
##### **Overview of the issue**

Compodoc duplicates properties and methods that are inherited from another classes/components.

##### **Operating System, Node.js, npm, compodoc version(s)**

 - macOS Big Sur: 11.2.1
 - node: v12.20.1
 - npm: 6.14.10
 - compodoc: 1.1.11

##### **Compodoc installed globally or locally ?**

`locally`

##### **Reproduce the error**

```ts
import { Component, Input } from '@angular/core';

@Component({ template: '' })
export abstract class MotherComponent {
  abstract myProp: string;
  abstract myInput: string;
  
  constructor() { }

  abstract myMethod(): string;
}

@Component({  template: '',})
export class SonComponent extends MotherComponent {
  myProp: string;
  
  @Input() myInput: string;
  
  constructor() {
    super();
  }

  myMethod(): string {
    return 'Implementation A';
  }
}

@Component({ template: '' })
export class GrandsonComponent extends SonComponent {
  constructor() {
    super();
  }

  myMethod(): string {
    return 'Implementation B';
  }
}
```

![image](https://user-images.githubusercontent.com/1134016/109189226-8e299900-7772-11eb-84f0-4b3edc19fbef.png)

![image](https://user-images.githubusercontent.com/1134016/109189261-97b30100-7772-11eb-94d5-182a652b22e8.png)

**Comment by armoucar** [comment]
WIth the pull-request, the result is:

![image](https://user-images.githubusercontent.com/1134016/109227171-aa432f80-779e-11eb-84d6-3358f0b94d27.png)

![image](https://user-images.githubusercontent.com/1134016/109227226-bcbd6900-779e-11eb-8a5b-7b08f686acea.png)

I think that the `Abstract` tag shouldn't be appearing on the `GransonComponent's myInput` because `SonComponent` already implemented it. But as the code is recursive to discover where the class member was first defined, it finds it on `MotherComponent`, that has the `Abstract` metadata bound to it.

**Comment by armoucar** [comment]
Hey @vogloblinsky, what do you think of this?

**Comment by vogloblinsky** [comment]
Hi @armoucar
Thanks for your time. Your code is just focused on component, but this inheritance process need to be generic for class, and all derivated : component, directive, injectables etc.
I close it and fix this issue soon before releasing 1.1.12
I will follow the natural process for reading such code with inheritance, and how Typedoc document this use-case too.
Regards
